### PR TITLE
[Patch] Making conversion not leave project on bad state

### PR DESCRIPTION
**Pull request by igor-ribeiiro**
Thank you for your pull request!  

Please add '[major]', '[minor]', or [patch] to the title to indicate the impact the change has on the code.  Please also provide the following information.

---

**Change Description**:

  Made the conversion check for any errors before making changes to the project (such as in the `packge.json`.

1. **Do these changes impact *command syntax* of any of the packages?** (e.g., add/remove command, add/remove a command parameter, or update required parameters)
No

2. **Do these changes impact *documentation*?** (e.g., a tutorial on https://docs.microsoft.com/en-us/office/dev/add-ins/overview/office-add-ins)
No

If you answered yes to any of these please do the following:
    > Include 'Rick-Kirkham' in the review
    > Make sure the README file is correct

**Validation/testing performed**:

Tested conversion of the project with an invalid manifest.

### CODE REVIEW

File: packages/office-addin-project/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "./lib/main.js",
   "scripts": {
     "build": "rimraf lib && concurrently \"tsc -p tsconfig.json\"",
-    "cli": "node lib/cli.js",
+    "cli": "node lib/cli.js convert -m test/invalid.manifest.xml",
```
Comment (millerds) [inline packages/office-addin-project/package.json:8]: I don't think we want to package.json script command to have specific arguments like this.  Those should come from the command line (I believe that is how the others work).

File: packages/office-addin-project/src/convert.ts [hunk 2]
```diff
@@ -37,9 +37,14 @@ export async function convertProject(
     return;
   }
   await backupProject(backupPath);
+  try {
+    await convert(manifestPath, outputPath, false, false);
+  } catch (err: any) {
+    console.log(`Error in conversion.`);
+    throw err;
+  }
```
Comment (millerds) [inline packages/office-addin-project/src/convert.ts:45]: Is this all we need to do?  Isn't there any clean up from a partially completed call to convert.

**Review by millerds — CHANGES_REQUESTED** [review]

File: packages/office-addin-project/src/convert.ts (hunk 2 above)
Comment (igor-ribeiiro) [reply packages/office-addin-project/src/convert.ts:45]: There would still be a `.zip` backup file that was created. Other than that. you shouldn't see other changes in the project (Just tested again and it is like that).

Comment (millerds) [reply packages/office-addin-project/src/convert.ts:45]: Right now that may be the case in our limited experience, but there may be something we miss or something that get's changed that alters this.  The backup may be there, but we're leaving it up to the customer to switch back even when we detect something went wrong.  It is a better customer experience to report the error, but them back into the state the started with and the way to be sure of that is to restore the backup for them.

File: packages/office-addin-project/src/convert.ts [hunk 3]
```diff
@@ -37,9 +37,15 @@ export async function convertProject(
...
   await backupProject(backupPath);
+  try {
+    await convert(manifestPath, outputPath, false, false);
+  } catch (err: any) {
+    console.log(`Error in conversion. Restoring project initial state.`);
+    await restoreBackup(backupPath);
+    throw err;
+  }
   updatePackages();
   await updateManifestXmlReferences();
```
Comment (millerds) [inline packages/office-addin-project/src/convert.ts:48]: These two lines should be part of the try statement.

**Review by millerds — CHANGES_REQUESTED** [review]

**Review by millerds — APPROVED** [review]

**Pull request merged**